Mathlib/MeasureTheory/Measure/SeparableMeasureSpace.lean

```diff
@@ -0,0 +1,455 @@
+        (biUnion_subset_biUnion_left fun k hkm ↦ Nat.le_trans hkm hmn)
+    have := tendsto_measure_iUnion (μ := μ) mono
+    rw [← tendsto_toReal_iff] at this
+    · -- We can therefore choose `N` such that `μ s - μ ((S N) ∩ s) < ε/2`.
+      rcases Metric.tendsto_atTop.1 this (ε / 2) (by linarith [ε_pos]) with ⟨N, hN⟩
+      have : Fact (μ (T N) < ∞) := Fact.mk <| T_finite N
+      -- Then we can apply the previous result to the measure `μ ((S N) ∩ •)`.
+      -- There exists `t ∈ 𝒜` such that `μ ((S N) ∩ (s ∆ t)) < ε/2`.
+      rcases (measureDense_of_generateFrom_setAglebra_of_finite
+        (μ := μ.restrict (T N)) h𝒜 hgen).approx s ms
+        (ne_of_lt (lt_of_le_of_lt (μ.restrict_apply_le _ s) hμs.lt_top))
+        (ε / 2) (by linarith [ε_pos])
+        with ⟨t, t_mem, ht⟩
+      -- We can then use `t ∩ (S N)`, because `S N ∈ 𝒜` by hypothesis.
+      -- `μ (s ∆ (t ∩ S N))`
+      --   `≤ μ (s ∆ (s ∩ S N)) + μ ((s ∩ S N) ∆ (t ∩ S N))`
+      --   `= μ s - μ (s ∩ S N) + μ (s ∆ t) ∩ S N) < ε`.
+      refine ⟨t ∩ T N, h𝒜.inter_mem t_mem (T_mem N), ?_⟩
+      calc
+        μ (s ∆ (t ∩ T N))
+          ≤ μ (s \ (s ∩ T N)) + μ ((s ∆ t) ∩ T N) := by
+              rw [← symmDiff_of_le (inter_subset_left ..), symmDiff_comm _ s,
+                inter_symmDiff_distrib_right]
+              exact measure_symmDiff_le _ _ _
+        _ < ENNReal.ofReal (ε / 2) + ENNReal.ofReal (ε / 2) := by
+              apply ENNReal.add_lt_add
+              · rw [measure_diff
+                    (inter_subset_left ..)
+                    (ms.inter (hgen ▸ measurableSet_generateFrom (T_mem N)))
+                    (ne_top_of_le_ne_top hμs (measure_mono (inter_subset_left ..))),
+                  lt_ofReal_iff_toReal_lt (sub_ne_top hμs),
+                  toReal_sub_of_le (measure_mono (inter_subset_left ..)) hμs]
+                apply lt_of_le_of_lt (sub_le_dist ..)
+                nth_rw 1 [← univ_inter s]
+                rw [inter_comm s, dist_comm, ← T_spanning, iUnion_inter _ T]
+                apply hN N (le_refl _)
+              · rwa [← μ.restrict_apply' (hgen ▸ measurableSet_generateFrom (T_mem N))]
+        _ = ENNReal.ofReal ε := by
+              rw [← ofReal_add (by linarith [ε_pos]) (by linarith [ε_pos]), add_halves]
+    · exact fun n ↦ ne_top_of_le_ne_top hμs (measure_mono (inter_subset_right ..))
+    · exact ne_top_of_le_ne_top hμs
+        (measure_mono (iUnion_subset (fun i ↦ inter_subset_right ..)))
+
+end MeasureDense
+
+section SeparableMeasureSpace
+
+/-! ### Definition of a separable measure space, sufficient condition -/
+
+/-- A measure space `X` is separable if it admits a countable and measure-dense family of sets.
+
+The term "separable" is justified by the fact that the definition translates to the usual notion
+of separability in the metric space made by constant indicators equipped with the `L¹` norm. -/
+class SeparableMeasureSpace (μ : Measure X) : Prop :=
```

Comment:
We have a specific meaning for `MeasureSpace` in mathlib: it is a measurable space endowed with a canonical measure, called `volume`. This is not what you are speaking about, so your name should not mention `MeasureSpace`. As it's a property of the measure, it should also start with `Is` if possible. What about `IsCountablySeparated`, for instance?

Reply:
I am a bit reluctant to use "separated" when talking about separability because I find it confusing with the notion of separation, e.g. [MeasurableSpace.SeparatesPoints](https://leanprover-community.github.io/mathlib4_docs/Mathlib/MeasureTheory/MeasurableSpace/CountablyGenerated.html#MeasurableSpace.SeparatesPoints). Maybe I can use `IsSeparable`? This is no longer used in #12394, it was replaced by `IsSeparablySupported`. Tell me what you think.

Reply:
ok for `IsSeparable`!